### PR TITLE
External mysql fix

### PR DESCRIPTION
**Pull request by dandersonsw**
Fixes the external mysql db code, and also fixes spec tests.

### CODE REVIEW

**Comment by dandersonsw** [comment]
This will need to be backported to v2.0 of the kit

**Review by dennisjbell — CHANGES_REQUESTED** [review]

File: overlay/addons/external-db.yml [hunk 1]
```diff
@@ -87,7 +86,6 @@ instance_groups:
               - name: (( grab params.uaa_db_name ))
                 tag: uaa
             roles:
-              - (( delete "postgres" ))
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply overlay/addons/external-db.yml]: fixed

File: overlay/addons/external-db-mysql.yml [hunk 2]
```diff
@@ -22,21 +24,20 @@ instance_groups:
...
+        roles:
+          - (( delete "postgres" ))
 
   - name: bbr-uaadb
     properties:
       uaadb:
         db_scheme: mysql
         port: 3306
+        roles:
+          - (( delete "postgres" ))
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply overlay/addons/external-db-mysql.yml]: Moved back to external-db

File: spec/results/all-addons-source.yml [hunk 3]
```diff
@@ -234,12 +240,12 @@ instance_groups:
       credhub:
         data_storage:
           database: credhub
-          host: 127.0.0.1
-          password: <!{meta.vault}/db:password!>
+          host: test-external-host
+          password: <!{meta.vault}/external-db/credhub_user:password!>
           port: 5432
-          require_tls: false
+          require_tls: true
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply spec/results/all-addons-source.yml]: now properly set to false

File: spec/results/all-addons-source.yml [hunk 4]
```diff
@@ -370,9 +376,12 @@ instance_groups:
...
         adapter: postgres
         database: bosh
-        host: 127.0.0.1
-        password: <!{meta.vault}/db:password!>
-        user: postgres
+        host: test-external-host
+        password: <!{meta.vault}/external-db/bosh_user:password!>
+        port: 5432
+        tls:
+          enabled: true
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply spec/results/all-addons-source.yml]: fixed

File: spec/results/all-addons.yml [hunk 5]
```diff
@@ -184,16 +184,22 @@ instance_groups:
     properties:
       release_level_backup: true
       uaadb:
-        address: 127.0.0.1
+        address: test-external-host
         databases:
         - name: uaa
           tag: uaa
-        db_scheme: postgresql
+        - name: uaa
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply spec/results/all-addons.yml]: Now only added once

File: spec/results/all-addons-source.yml [hunk 6]
```diff
@@ -182,16 +182,22 @@ instance_groups:
...
         databases:
         - name: uaa
           tag: uaa
-        db_scheme: postgresql
+        - name: uaa
+          tag: uaa
+        db_scheme: postgres
         port: 5432
         roles:
         - name: postgres
```
[the remark this answers is not in the document]
Comment (dandersonsw) [reply spec/results/all-addons-source.yml]: No longer there when external-db is used

File: overlay/addons/external-db-no-tls.yml [hunk 7]
```diff
@@ -1,12 +1,19 @@
...
+      enabled: false
+
 instance_groups:
   - name: bosh
     properties:
       director:
         db:
           tls:
             enabled: false
+      registry:
```
Comment (dennisjbell) [inline overlay/addons/external-db-no-tls.yml:15]: This isn't needed in this file, the one in external-db.yml will suffice

File: hooks/blueprint [hunk 8]
```diff
@@ -283,7 +283,9 @@ if [[ "${version}" != "" ]] && ! new_enough "${version}" "1.15.2"; then
 fi
 # this is only used for upgrading from 1.15.2 to 2.0 and can be removed in later versions
 if [[ "${version}" != "" ]] && ! new_enough "${version}" "2.0" ; then
- merge+=( overlay/addons/change-db-owner.yml )
+ if ! want_feature "external-db-mysql" ; then
+  merge+=( overlay/addons/change-db-owner.yml )
+ fi
 fi
```
Comment (dennisjbell) [inline hooks/blueprint:289]: No sense nesting ifs...

```
if [[ "${version}" != "" ]] && ! new_enough "${version}" "2.0"  && ! want_feature "external-db-mysql" ; then
  merge+=( overlay/addons/change-db-owner.yml )
fi
```

**Review by dennisjbell — CHANGES_REQUESTED** [review]

File: overlay/addons/external-db-no-tls.yml (hunk 7 above)
Comment (dandersonsw) [reply overlay/addons/external-db-no-tls.yml]: removed

File: hooks/blueprint (hunk 8 above)
Comment (dandersonsw) [reply hooks/blueprint:287]: updated with no nested ifs and using "+external-db"

**Pull request merged**